packages/property-view/src/browser/resource-property-view/resource-property-data-service.spec.ts

```diff
@@ -0,0 +1,139 @@
+/********************************************************************************
+ * Copyright (C) 2021 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { enableJSDOM } from '@theia/core/lib/browser/test/jsdom';
+
+let disableJSDOM = enableJSDOM();
+
+import { expect } from 'chai';
+import { FrontendApplicationConfigProvider } from '@theia/core/lib/browser/frontend-application-config-provider';
+import { ApplicationProps } from '@theia/application-package/lib/application-props';
+FrontendApplicationConfigProvider.set({
+    ...ApplicationProps.DEFAULT.frontend.config
+});
+
+import { Container } from '@theia/core/shared/inversify';
+import { ResourcePropertyDataService } from './resource-property-data-service';
+import { FileService } from '@theia/filesystem/lib/browser/file-service';
+import URI from '@theia/core/lib/common/uri';
+import { PropertyDataService } from '../property-data-service';
+import { FileSelection } from '@theia/filesystem/lib/browser/file-selection';
+import { Navigatable } from '@theia/core/lib/browser/navigatable';
+import { FileStat } from '@theia/filesystem/lib/common/files';
+
+disableJSDOM();
+
+let resourcePropertyDataService: ResourcePropertyDataService;
+
+const MockFileStat: FileStat = {
+    isFile: false,
+    isDirectory: true,
+    isSymbolicLink: false,
+    resource: new URI('resource'),
+    name: 'name'
+};
+
+describe('resource-property-data-service', () => {
+
+    before(() => {
+        disableJSDOM = enableJSDOM();
+        const container = new Container();
+        container.bind(ResourcePropertyDataService).toSelf().inSingletonScope();
+        container.bind(FileService).toConstantValue({
+            async resolve(_uri: URI): Promise<FileStat> {
+                MockFileStat.resource = _uri;
+                return MockFileStat;
+            }
```

Comment:
Since `_uri` used there is no need for the trailing `_`:

```suggestion
            async resolve(uri: URI): Promise<FileStat> {
                mockFileStat.resource = uri;
                return mockFileStat;
            }
```

I'm not really sure why you want to mutate the mock with the `resolve` like you're doing and not simply do:

```ts
async resolve(_uri: URI): Promise<FileStat> {
  return mockFileStat;
}
```

Reply:
I did that at the beginning because I wanted to test if the uri of the FileStat that was returned by the `providePropertyData` method was equal to the uri of the file selection (or navigatable selection). I forgot to remove it after modifying the tests.